src/game/shell.c

```diff
@@ -187,6 +189,60 @@ void Shell_Wait(int nticks)
     }
 }
 
+void Shell_ValidateLevelTitle(char *str)
+{
+    // Strip non alphanumeric chars
+    char *ix = str;
+    int idx = 0;
+    while (*ix != '\0') {
+        if ((*ix >= 'A' && *ix <= 'Z') || (*ix >= 'a' && *ix <= 'z')
+            || (*ix >= '0' && *ix <= '9') || *ix == '\'' || *ix == '.'
+            || *ix == ' ') {
+            idx++;
+            *ix++;
+        } else {
+            memmove(&str[idx], &str[idx + 1], strlen(str) - idx);
+        }
+    }
+
+    // Make sure title between 0 and LEVEL_TITLE_SIZE chars
+    if (strlen(str) == 0) {
+        memmove(str + 1, str, 2);
+        str[0] = 'x';
+        return;
+    } else if (strlen(str) > LEVEL_TITLE_SIZE) {
+        str[LEVEL_TITLE_SIZE] = '\0';
+    }
+}
+
+void Shell_GetScreenshotName(char *str)
+{
+    // Replace spaces with underscores
+    char *check = str;
+    while ((check = strchr(check, ' ')) != NULL) {
+        *check++ = '_';
+    }
+
+    // Remove special chars from screenshots that are allowed in titles
+    int idx = 0;
+    check = str;
+    while (*check != '\0') {
+        if (*check == '\'' || *check == '.') {
+            memmove(&str[idx], &str[idx + 1], strlen(str) - idx);
+        } else {
+            idx++;
+            *check++;
+        }
+    }
+
+    // Make sure title has at least 1 char
+    if (strlen(str) == 0) {
+        memmove(str + 1, str, 2);
+        str[0] = 'x';
+        return;
+    }
+}
```

Comment:
I'd expect this function to return a complete screenshot name and not have to amend its return value in the client code later. In other words I'd make sure this functions does the `sprintf("%s_%s.%s", date_time, level_title, ext)` part.

I'd also allocate memory here and free it in the client code (where we use this function) via `Memory_Free`.

So something like `char *Shell_GetScreenshotName(const char *level_title, const char *ext)` where `level_title` is the unprocessed level title.